♻️(lti_toolbox) Enhance code readability base LTI view classes

lebaudantoine opened the pull request:
## Purpose
Avoid any confusion for newcomers to the project.

## Proposition
- [x] Use `@abstractmethod` decorator when needed to make code consistent.
- [x] Update classes' documentation

lebaudantoine committed:
♻️(lti_toolbox) use modern python decorator in `BaseLTIView`

Make use of the right method decorator '@abstractmethod' to make clear
'_do_on_success' method must be implemented by subclasses.

lebaudantoine committed:
📝(lti_toolbox) clarify differences between base view classes

The documentation for 'BaseLTIView' was mistakenly copied
from 'BaseLTIAuthViews,' potentially causing confusion for
newcomers to the project. This commit rectifies the issue by
updating the 'BaseLTIView' documentation to accurately reflect
its unique functionality and purpose.

lebaudantoine committed:
✏️(changelog) fix a minor typo in the 1.0.0b1 Changelog

Nit-picking on the added features Changelog description.

wilbrdt reviewed (approved):
![Alt text](https://media.giphy.com/media/lcySndwSDLxC4eOU86/giphy.gif)
LGTM, mergeable after pleasing the CI!

lebaudantoine committed:
⬆️(dependency) upgrade black linter to 22.3.0

Updated black and fixed consequent issues.

lebaudantoine committed:
💚(ci) fix gitmoji json file url

Fixed lint-git CI job that was failing. The json file, which describes
allowed gitmoji, has been moved.

lebaudantoine committed:
💚(ci) pin Django to a version < 4.2

Fixed test CI job was failing, migrating to Django 4.2 causes some
dependencies issues. These happen because some Django settings
have been deprecated.
